__tests__/commands/install/integration.js

```diff
@@ -102,36 +103,43 @@ test.concurrent('hoisting should factor ignored dependencies', async () => {
       (await fs.readJson(path.join(config.cwd, 'node_modules', 'd', 'package.json'))).version,
       '1.0.0',
     );
-
-    assert.equal(
-      (await fs.readJson(path.join(config.cwd, 'node_modules', 'd', 'node_modules', 'c', 'package.json'))).version,
-      '2.0.0',
-    );
   }
 
   await runInstall({}, 'install-ignored-retains-hoisting-structure', async (config) => {
     await checkNormal(config);
 
+    // devDependecy b should be installed
     assert.equal(
       (await fs.readJson(path.join(config.cwd, 'node_modules', 'b', 'package.json'))).version,
       '3.0.0',
     );
 
+    // c is depended upon in both 5.0.0 (through b) and 2.0.0 (through d)
+    // 5.0.0 should be hoisted
+    // while 2.0.0 should be installed for dependent package d
     assert.equal(
       (await fs.readJson(path.join(config.cwd, 'node_modules', 'c', 'package.json'))).version,
       '5.0.0',
     );
+
+    assert.equal(
+      (await fs.readJson(path.join(config.cwd, 'node_modules', 'd', 'node_modules', 'c', 'package.json'))).version,
+      '2.0.0',
+    );
   });
 
   await runInstall({production: true}, 'install-ignored-retains-hoisting-structure', async (config) => {
     await checkNormal(config);
 
+    // devDependency b should not be installed
     assert.ok(
       !await fs.exists(path.join(config.cwd, 'node_modules', 'b')),
     );
 
-    assert.ok(
-      !await fs.exists(path.join(config.cwd, 'node_modules', 'c')),
+    // c is only depended upon by d in version 2.0.0, so it should be hoisted
+    assert.equal(
+      (await fs.readJson(path.join(config.cwd, 'node_modules', 'c', 'package.json'))).version,
```

Comment:
That is the idea of Yarn - it keeps packages hoisting deterministic between normal and production build.
The test checked that hoisting of C should be stable and I think it was correct